src/Tribe/Ticket_Object.php

```diff
@@ -263,17 +284,83 @@ public function end_date() {
 		 *
 		 * @return boolean Whether or not the provided date/time falls within the start/end date range
 		 */
-		public function date_in_range( $datetime ) {
-			if ( is_numeric( $datetime ) ) {
-				$timestamp = $datetime;
-			} else {
-				$timestamp = strtotime( $datetime );
+		public function date_in_range( $datetime = 'now' ) {
+			$timestamp = is_numeric( $datetime ) ? $datetime : strtotime( $datetime );
+			// Attempt to convert the timestamp to a Date object.
+			try {
+				$timezone = $this->get_event_timezone();
+				if ( 'now' === $datetime ) {
+					$now = new DateTime( 'now', $timezone  );
+				} else {
+					$now = new DateTime( '@' . $timestamp );
+					if ( $timezone instanceof DateTimeZone ) {
+						$now->setTimezone( $timezone );
+					}
+				}
+			} catch ( Exception $exception ) {
+				return false;
 			}
 
-			$start_date = $this->start_date();
-			$end_date   = $this->end_date();
+			$start = $this->start_date( false );
+			$end = $this->end_date( false );
+
+			if ( ! $start instanceof DateTime || ! $end instanceof DateTime ) {
+				$now = $timestamp;
+			}
+
+			return ( empty( $start ) || $now >= $start ) && ( empty( $end ) || $now <= $end );
+		}
+
+
+		/**
+		 * Get a DateTime object or a Timestamp date representation of the object, if the DateTime object is used
+		 * the timezone from the event associated with the ticket is going to be used to have a more accurate
+		 * timestamp
+		 *
+		 * @since TBD
+		 *
+		 * @param string $date
+		 * @param bool $as_timestamp
+		 *
+		 * @return DateTime|false|int
+		 */
+		public function get_date( $date = '', $as_timestamp = true ) {
+
+			if ( $as_timestamp ) {
+				return strtotime( $date );
+			}
+
+			try {
+				$timezone = $this->get_event_timezone();
+				return new DateTime( $date, $timezone );
+			} catch ( Exception $exception ) {
+				return strtotime( $date );
+			}
+		}
+
+
+		/**
+		 * Return a DateTimeZone associated with the parent Event of the current ticket
+		 *
+		 * @since TBD
+		 *
+		 * @return DateTimeZone|null
+		 */
+		public function get_event_timezone() {
+
+			if ( is_null( $this->event_id ) ) {
+				$this->event_id = $this->get_event();
```

Comment:
`get_event()` will fetch the whole object, we should either save the whole object in memory, but I would rather see it been fetched when you hit the `get_event()` method instead.

Reply:
`getter` has been added 👍